consensus/consensus-types/src/executed_block.rs

```diff
@@ -162,46 +162,19 @@ impl ExecutedBlock {
         validators: &[AccountAddress],
         validator_txns: Vec<ValidatorTransaction>,
         txns: Vec<SignedTransaction>,
-        is_block_gas_limit: bool,
     ) -> Vec<Transaction> {
         // reconfiguration suffix don't execute
 
         if self.is_reconfiguration_suffix() {
             return vec![];
         }
 
-        let mut txns_with_state_checkpoint = self.block.transactions_to_execute(
-            validators,
-            validator_txns,
-            txns,
-            is_block_gas_limit,
-        );
-        if is_block_gas_limit && !self.state_compute_result.has_reconfiguration() {
-            // After the per-block gas limit change,
-            // insert state checkpoint at the position
-            // 1) after last txn if there is no Retry
-            // 2) before the first Retry
-            if let Some(pos) = self
-                .state_compute_result
-                .compute_status()
-                .iter()
-                .position(|s| s.is_retry())
-            {
-                txns_with_state_checkpoint.insert(pos, Transaction::StateCheckpoint(self.id()));
-            } else {
-                txns_with_state_checkpoint.push(Transaction::StateCheckpoint(self.id()));
-            }
-        }
+        let input_txns = self
```

Comment:
Add some comment for future readers? like `input_txns` contains blockmetadata, and `transactions_to_commit` will add state checkpoint.